go/tasks/pluginmachinery/flytek8s/config/config.go

```diff
@@ -128,9 +130,17 @@ type K8sPluginConfig struct {
 	CreateContainerErrorGracePeriod config2.Duration `json:"create-container-error-grace-period" pflag:"-,Time to wait for transient CreateContainerError errors to be resolved."`
 
 	// The name of the GPU resource to use when the task resource requests GPUs.
-	GpuResourceName v1.ResourceName `json:"gpu-resource-name" pflag:",The name of the GPU resource to use when the task resource requests GPUs."`
+	GpuResourceName v1.ResourceName `json:"gpu-resource-name" pflag:"-,The name of the GPU resource to use when the task resource requests GPUs."`
+
+	// DefaultPodSecurityContext provides a default pod security context that should be applied for every pod that is launched by FlytePropeller
+	DefaultPodSecurityContext *v1.PodSecurityContext `json:"default-pod-security-context" pflag:"-,Optionally specify any default pod security context that should be applied to every Pod launched by FlytePropeller."`
+
+	// DefaultSecurityContext provides a default container security context that should be applied for every container launched by FlytePropeller
+	DefaultSecurityContext *v1.SecurityContext `json:"default-security-context" pflag:"-,Optionally specify a default security context that should be applied to every container launched by FlytePropeller."`
```

Comment:
do we really need both if this is anyways defaulted at platform level right now ? The PodSecurityContext will anyways be applied to all containers in every pod created by FlytePropeller ?

Reply:
is that really true?

Reply:
So i guess having both has one benefit, you can simply add ContainerSecurityContext and that will be applied only to your container not to the injected containers. While podSecuritycontext will be applied to everything.
If you specify both, then the resultant will be a combination. WDYT?

Reply:
I guess that makes sense. Having both will probably cause some confusion/raise questions so let's  document the overlap in behavior here.